upstream-sha256: b894100bf9b5e0ba6d812372a8ec231571c2360ea10db33278496cdc6564031c
actual-sha256: f9ec5144e56440e479686b283529317aada0da3ed84ec23ce5bc86ca1409208c
--- upstream.expected
+++ tolk.actual
@@ -172,80 +172,82 @@
   67 Ops.CAST            :            dtypes.uchar                             ['3']                            dtypes.uchar
   68 Ops.SHR             :            dtypes.uchar                             [66, 67]                         None
   69 Ops.CAST            :            dtypes.float                             [68]                             dtypes.float
-  70 Ops.MUL             :            dtypes.float                             [69, 56]                         None
-  71 Ops.ADD             :            dtypes.float                             [63, 70]                         None
-  72 Ops.MUL             :            dtypes.float                             [61, 71]                         None
-  73 Ops.PARAM           :            dtypes.float                             []                               ParamArg(3, dtypes.float, 12, device='CPU')
-  74 Ops.SHL             :            dtypes.int                               [3, 11]                          None
-  75 Ops.SHR             :            dtypes.int                               [9, 38]                          None
-  76 Ops.ADD             :            dtypes.int                               [74, 75]                         None
-  77 Ops.INDEX           :            dtypes.float                             [73, 76]                         None
-  78 Ops.LOAD            :            dtypes.float                             [77]                             None
-  79 Ops.MUL             :            dtypes.float                             [72, 78]                         None
-  80 Ops.CAST            :            dtypes.uchar                             ['4']                            dtypes.uchar
-  81 Ops.SHR             :            dtypes.uchar                             [41, 80]                         None
-  82 Ops.AND             :            dtypes.uchar                             [81, 43]                         None
-  83 Ops.CAST            :            dtypes.float                             [82]                             dtypes.float
-  84 Ops.CMPLT           :            dtypes.bool                              [83, 47]                         None
-  85 Ops.MUL             :            dtypes.float                             [83, 50]                         None
-  86 Ops.CMPLT           :            dtypes.bool                              [83, 53]                         None
-  87 Ops.ADD             :            dtypes.float                             [83, 56]                         None
-  88 Ops.WHERE           :            dtypes.float                             [86, 87, 59]                     None
-  89 Ops.WHERE           :            dtypes.float                             [84, 85, 88]                     None
-  90 Ops.SHR             :            dtypes.uchar                             [81, 67]                         None
-  91 Ops.CAST            :            dtypes.float                             [90]                             dtypes.float
-  92 Ops.MUL             :            dtypes.float                             [91, 56]                         None
-  93 Ops.ADD             :            dtypes.float                             [63, 92]                         None
-  94 Ops.MUL             :            dtypes.float                             [89, 93]                         None
-  95 Ops.MUL             :            dtypes.float                             [94, 78]                         None
-  96 Ops.STACK           :            dtypes.float                             [79, 95]                         None
-  97 Ops.STORE           :            dtypes.void                              [16, 96]                         None
-  98 Ops.CONST           :            dtypes.weakint                           []                               192
-  99 Ops.CAST            :            dtypes.int                               ['192']                          dtypes.int
- 100 Ops.ADD             :            dtypes.int                               [13, 99]                         None
- 101 Ops.SHRINK          :            dtypes.float                             [0, 100, 15]                     None
- 102 Ops.INDEX           :            dtypes.int                               [26, 11]                         None
- 103 Ops.MUL             :            dtypes.int                               [102, 29]                        None
- 104 Ops.ADD             :            dtypes.int                               [21, 103]                        None
- 105 Ops.INDEX           :            dtypes.uchar                             [17, 104]                        None
- 106 Ops.CMPLT           :            dtypes.bool                              [35, 102]                        None
- 107 Ops.CMPLT           :            dtypes.bool                              [102, 38]                        None
- 108 Ops.AND             :            dtypes.bool                              [106, 107]                       None
- 109 Ops.LOAD            :            dtypes.uchar                             [105, 33, 108]                   None
- 110 Ops.AND             :            dtypes.uchar                             [109, 43]                        None
- 111 Ops.CAST            :            dtypes.float                             [110]                            dtypes.float
- 112 Ops.CMPLT           :            dtypes.bool                              [111, 47]                        None
- 113 Ops.MUL             :            dtypes.float                             [111, 50]                        None
- 114 Ops.CMPLT           :            dtypes.bool                              [111, 53]                        None
- 115 Ops.ADD             :            dtypes.float                             [111, 56]                        None
- 116 Ops.WHERE           :            dtypes.float                             [114, 115, 59]                   None
- 117 Ops.WHERE           :            dtypes.float                             [112, 113, 116]                  None
- 118 Ops.AND             :            dtypes.uchar                             [109, 65]                        None
- 119 Ops.SHR             :            dtypes.uchar                             [118, 67]                        None
- 120 Ops.CAST            :            dtypes.float                             [119]                            dtypes.float
- 121 Ops.MUL             :            dtypes.float                             [120, 56]                        None
- 122 Ops.ADD             :            dtypes.float                             [63, 121]                        None
- 123 Ops.MUL             :            dtypes.float                             [117, 122]                       None
- 124 Ops.ADD             :            dtypes.int                               [76, 5]                          None
- 125 Ops.INDEX           :            dtypes.float                             [73, 124]                        None
- 126 Ops.LOAD            :            dtypes.float                             [125]                            None
- 127 Ops.MUL             :            dtypes.float                             [123, 126]                       None
- 128 Ops.SHR             :            dtypes.uchar                             [109, 80]                        None
- 129 Ops.AND             :            dtypes.uchar                             [128, 43]                        None
- 130 Ops.CAST            :            dtypes.float                             [129]                            dtypes.float
- 131 Ops.CMPLT           :            dtypes.bool                              [130, 47]                        None
- 132 Ops.MUL             :            dtypes.float                             [130, 50]                        None
- 133 Ops.CMPLT           :            dtypes.bool                              [130, 53]                        None
- 134 Ops.ADD             :            dtypes.float                             [130, 56]                        None
- 135 Ops.WHERE           :            dtypes.float                             [133, 134, 59]                   None
- 136 Ops.WHERE           :            dtypes.float                             [131, 132, 135]                  None
- 137 Ops.SHR             :            dtypes.uchar                             [128, 67]                        None
- 138 Ops.CAST            :            dtypes.float                             [137]                            dtypes.float
- 139 Ops.MUL             :            dtypes.float                             [138, 56]                        None
- 140 Ops.ADD             :            dtypes.float                             [63, 139]                        None
- 141 Ops.MUL             :            dtypes.float                             [136, 140]                       None
- 142 Ops.MUL             :            dtypes.float                             [141, 126]                       None
- 143 Ops.STACK           :            dtypes.float                             [127, 142]                       None
- 144 Ops.STORE           :            dtypes.void                              [101, 143]                       None
- 145 Ops.GROUP           :            dtypes.void                              [97, 144, 97, 144]               None
- 146 Ops.SINK            :            dtypes.void                              [145]                            KernelInfo(name='E_3_32_2_2', applied_opts=(Opt(op=OptOps.SPLIT, axis=2, arg=(0, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(0, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(32, AxisType.LOCAL))), opts_to_apply=None, estimates=None, beam=0)
+  70 Ops.CONST           :            dtypes.weakfloat                         []                               2.0
+  71 Ops.CAST            :            dtypes.float                             ['2.0']                          dtypes.float
+  72 Ops.MUL             :            dtypes.float                             [69, 71]                         None
+  73 Ops.SUB             :            dtypes.float                             [63, 72]                         None
+  74 Ops.MUL             :            dtypes.float                             [61, 73]                         None
+  75 Ops.PARAM           :            dtypes.float                             []                               ParamArg(3, dtypes.float, 12, device='CPU')
+  76 Ops.SHL             :            dtypes.int                               [3, 11]                          None
+  77 Ops.SHR             :            dtypes.int                               [9, 38]                          None
+  78 Ops.ADD             :            dtypes.int                               [76, 77]                         None
+  79 Ops.INDEX           :            dtypes.float                             [75, 78]                         None
+  80 Ops.LOAD            :            dtypes.float                             [79]                             None
+  81 Ops.MUL             :            dtypes.float                             [74, 80]                         None
+  82 Ops.CAST            :            dtypes.uchar                             ['4']                            dtypes.uchar
+  83 Ops.SHR             :            dtypes.uchar                             [41, 82]                         None
+  84 Ops.AND             :            dtypes.uchar                             [83, 43]                         None
+  85 Ops.CAST            :            dtypes.float                             [84]                             dtypes.float
+  86 Ops.CMPLT           :            dtypes.bool                              [85, 47]                         None
+  87 Ops.MUL             :            dtypes.float                             [85, 50]                         None
+  88 Ops.CMPLT           :            dtypes.bool                              [85, 53]                         None
+  89 Ops.ADD             :            dtypes.float                             [85, 56]                         None
+  90 Ops.WHERE           :            dtypes.float                             [88, 89, 59]                     None
+  91 Ops.WHERE           :            dtypes.float                             [86, 87, 90]                     None
+  92 Ops.SHR             :            dtypes.uchar                             [83, 67]                         None
+  93 Ops.CAST            :            dtypes.float                             [92]                             dtypes.float
+  94 Ops.MUL             :            dtypes.float                             [93, 71]                         None
+  95 Ops.SUB             :            dtypes.float                             [63, 94]                         None
+  96 Ops.MUL             :            dtypes.float                             [91, 95]                         None
+  97 Ops.MUL             :            dtypes.float                             [96, 80]                         None
+  98 Ops.STACK           :            dtypes.float                             [81, 97]                         None
+  99 Ops.STORE           :            dtypes.void                              [16, 98]                         None
+ 100 Ops.CONST           :            dtypes.weakint                           []                               192
+ 101 Ops.CAST            :            dtypes.int                               ['192']                          dtypes.int
+ 102 Ops.ADD             :            dtypes.int                               [13, 101]                        None
+ 103 Ops.SHRINK          :            dtypes.float                             [0, 102, 15]                     None
+ 104 Ops.INDEX           :            dtypes.int                               [26, 11]                         None
+ 105 Ops.MUL             :            dtypes.int                               [104, 29]                        None
+ 106 Ops.ADD             :            dtypes.int                               [21, 105]                        None
+ 107 Ops.INDEX           :            dtypes.uchar                             [17, 106]                        None
+ 108 Ops.CMPLT           :            dtypes.bool                              [35, 104]                        None
+ 109 Ops.CMPLT           :            dtypes.bool                              [104, 38]                        None
+ 110 Ops.AND             :            dtypes.bool                              [108, 109]                       None
+ 111 Ops.LOAD            :            dtypes.uchar                             [107, 33, 110]                   None
+ 112 Ops.AND             :            dtypes.uchar                             [111, 43]                        None
+ 113 Ops.CAST            :            dtypes.float                             [112]                            dtypes.float
+ 114 Ops.CMPLT           :            dtypes.bool                              [113, 47]                        None
+ 115 Ops.MUL             :            dtypes.float                             [113, 50]                        None
+ 116 Ops.CMPLT           :            dtypes.bool                              [113, 53]                        None
+ 117 Ops.ADD             :            dtypes.float                             [113, 56]                        None
+ 118 Ops.WHERE           :            dtypes.float                             [116, 117, 59]                   None
+ 119 Ops.WHERE           :            dtypes.float                             [114, 115, 118]                  None
+ 120 Ops.AND             :            dtypes.uchar                             [111, 65]                        None
+ 121 Ops.SHR             :            dtypes.uchar                             [120, 67]                        None
+ 122 Ops.CAST            :            dtypes.float                             [121]                            dtypes.float
+ 123 Ops.MUL             :            dtypes.float                             [122, 71]                        None
+ 124 Ops.SUB             :            dtypes.float                             [63, 123]                        None
+ 125 Ops.MUL             :            dtypes.float                             [119, 124]                       None
+ 126 Ops.ADD             :            dtypes.int                               [78, 5]                          None
+ 127 Ops.INDEX           :            dtypes.float                             [75, 126]                        None
+ 128 Ops.LOAD            :            dtypes.float                             [127]                            None
+ 129 Ops.MUL             :            dtypes.float                             [125, 128]                       None
+ 130 Ops.SHR             :            dtypes.uchar                             [111, 82]                        None
+ 131 Ops.AND             :            dtypes.uchar                             [130, 43]                        None
+ 132 Ops.CAST            :            dtypes.float                             [131]                            dtypes.float
+ 133 Ops.CMPLT           :            dtypes.bool                              [132, 47]                        None
+ 134 Ops.MUL             :            dtypes.float                             [132, 50]                        None
+ 135 Ops.CMPLT           :            dtypes.bool                              [132, 53]                        None
+ 136 Ops.ADD             :            dtypes.float                             [132, 56]                        None
+ 137 Ops.WHERE           :            dtypes.float                             [135, 136, 59]                   None
+ 138 Ops.WHERE           :            dtypes.float                             [133, 134, 137]                  None
+ 139 Ops.SHR             :            dtypes.uchar                             [130, 67]                        None
+ 140 Ops.CAST            :            dtypes.float                             [139]                            dtypes.float
+ 141 Ops.MUL             :            dtypes.float                             [140, 71]                        None
+ 142 Ops.SUB             :            dtypes.float                             [63, 141]                        None
+ 143 Ops.MUL             :            dtypes.float                             [138, 142]                       None
+ 144 Ops.MUL             :            dtypes.float                             [143, 128]                       None
+ 145 Ops.STACK           :            dtypes.float                             [129, 144]                       None
+ 146 Ops.STORE           :            dtypes.void                              [103, 145]                       None
+ 147 Ops.GROUP           :            dtypes.void                              [99, 146, 99, 146]               None
+ 148 Ops.SINK            :            dtypes.void                              [147]                            KernelInfo(name='E_3_32_2_2', applied_opts=(Opt(op=OptOps.SPLIT, axis=2, arg=(0, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(0, AxisType.UPCAST)), Opt(op=OptOps.SPLIT, axis=0, arg=(32, AxisType.LOCAL))), opts_to_apply=None, estimates=None, beam=0)
